### PR TITLE
FinalDraft

### PR DESCRIPTION
**Pull request by d-delanna**
I think this will better explain what I was trying to say. Is it a common practice for people to create files for all of the modules that they use like I did below? I understand that I had to do this on PyCharm, but when I am trying to share it on GitHub, should I try to keep everything in one folder, or spread it out?

I did my last couple of changes for hangman.py. Feel free to continue to critique it if you want so I can use your comments for future references, but this will be my last draft. 

In the last block of code of hangman.py (starting on line 114), Instead of trying to understand Class, I just stuck both of the smaller functions inside of choose_method( ). It allowed for my code to still run properly, but when the game was over, it would return an error which I just ignored (using line 152) . I'd imagine that this probably isn't the right way to go about it so please tell me how you would do it. 

PS I forwarded you a meme in your email. Doesn't exactly pertain to me, but I thought it was pretty funny because none of my personal projects are ever going to be seen by anyone either.

### CODE REVIEW

**Comment by d-delanna** [comment]
I forgot to also mention that I started doing something with __main__, but decided to leave it for later projects. You can ignore what I left it as

**Comment by mabunday** [comment]
Ah OK, I understand your question now. How people organize their project(s) is always going to vary. There are standards and conventions in place, but they depend on the language, personal vs. professional project, platform (mobile [android vs. ios], desktop (mac vs. windows, etc.]), etc. etc. etc. I'll link you some resources/discussions on the subject for Python. You'll see some common themes between them, but they'll also have some differing suggestions and that's totally fine. Just pick something reasonable; there's no single "right" answer, but there are sensible answers. 

* https://realpython.com/python-application-layouts/
* https://docs.python-guide.org/writing/structure/ (goes with: https://github.com/navdeep-G/samplemod)
* https://github.com/pypa/sampleproject (slightly different, but similar!) 
* https://dev.to/codemouse92/dead-simple-python-project-structure-and-imports-38c6

For your project you have multiple projects within your git repository. This is fine and common! As they get bigger, you may want to consider having a repository for *each project*, which is what most of the examples you'll see about show. So instead of `https://github.com/d-delanna/PersonalProjects/hangman` you'd have https://github.com/d-delanna/hangman`. For now, I would recommend just keeping everything in the same GitHub repository for simplicity of tracking. 

Here's how I would recommend reorganizing your folder structure, but also read the articles I linked above and make your own decisions, too: 

```
PersonalProjects/
├── README.md
├── Hangman/
│   ├── .gitignore
│   ├── LICENSE
│   ├── README.md
│   ├── requirements.txt
│   ├── setup.py
│   ├── src/
│       ├── __init__.py
│       ├── hangman.py
│       ├── draw_vers3.py
│   ├── data/
│       ├── __init__.py
│       ├── wordbank_vers3.py
|   ├── tests/
│       ├── __init__.py
│
├── ASLMachineLearning/
│   ├── .gitignore
│   ├── LICENSE
│   ├── README.md
│   ├── requirements.txt
│   ├── setup.py
│   ├── src/
│       ├── __init__.py
│       ├── ASLMachineLearning.py 
│   ├── data/
│       ├── __init__.py
|   ├── tests/
│       ├── __init__.py
│
├── DeterminantCalculator/
│   ├── .gitignore
│   ├── LICENSE
│   ├── README.md
│   ├── requirements.txt
│   ├── __init__.py
│   ├── src/
│       ├── __init__.py
│       ├── determinant-calculator.py
|   ├── tests/
│       ├── __init__.py
```

Again, this is roughly just how I would set things up based on their current state. One suggestion: When you create these folders and move your files around, create the folder and move them within PyCharm (in the left pane file browser thing). It will ask you when you move files if you want to move "references" if you use imports anywhere. If you say yes, it will automatically update the files you reference (import) based on their new locations, otherwise you're going to have to update them yourself.

File: hangman.py [hunk 1]
```diff
@@ -108,125 +100,54 @@ def guess_letter(word):
                 print(''.join(blanks))
             __replace_character(guess, '-', alphabet, alphabet)
 
-        else:
+        else:  # if guess not valid letter
```
Comment (mabunday) [inline hangman.py]: I like that you're leaving comments on what the else conditions do :)

File: hangman.py [hunk 2]
```diff
@@ -1,74 +1,67 @@
...
-    
-    blanks = list('-' * len(word))
-    # I tried writing it in the format that you said, but because
-    # this is in a function, 'characters' is an 'unused variable'.
-    # I thought of just having it return 'characters' at the end, but 
-    # I'm not sure if that would really be helping anything
-    
-    
+        while character in phrase:
+            __replace_character(character, character, phrase, blanks)
```
Comment (mabunday) [inline hangman.py]: You might be interested in the built-in str.replace() method. 

https://www.geeksforgeeks.org/python-string-replace/

https://docs.python.org/3/library/stdtypes.html#str.replace (Other useful built-in methods I use a lot are `.strip()`, `.upper()/lower()`, `startswith()/endswith()`)

**Pull request closed without merging**